File: abc/abc143/abc143_d.rb
N = gets.to_i
L = gets.strip.split.map(&:to_i)

L.sort!
result = 0
(0..N - 3).each do |i|
  (i + 1..N - 2).each do |j|
    t = L[i] + L[j]
    idx = L.bsearch_index { |x| x >= t }
    result += (idx ? idx : L.size) - j - 1
  end
end
puts result

File: abc/abc143/abc143_d.py
# 二分探索
from bisect import bisect_left

N = int(input())
L = list(map(int, input().split()))

L.sort()
result = 0
for i in range(N - 2):
    for j in range(i + 1, N - 1):
        result += bisect_left(L, L[i] + L[j]) - j - 1
print(result)
